master/internal/trial.go

```diff
@@ -663,17 +662,21 @@ func mustParseTrialRunID(child *actor.Ref) int {
 }
 
 func (t *trial) maybeRestoreAllocation(ctx *actor.Context) (*model.Allocation, error) {
-	if !t.restored || !config.IsReattachEnabled() {
+	if !t.restored || !t.rm.IsReattachEnabled(ctx) {
 		return nil, nil
 	}
 
-	// Do we have an open allocation?
 	var allocations []model.Allocation
-	err := db.Bun().NewSelect().Model(&allocations).
+	selectQuery := db.Bun().NewSelect().Model(&allocations).
 		Where("task_id = ?", t.taskID).
-		Where("start_time IS NOT NULL").
-		Where("end_time IS NULL").
-		Scan(context.TODO())
+		Where("end_time IS NULL")
+
+	if t.rm.IsReattachableOnlyAfterStarted(ctx) {
```

Comment:
one thing we have to take into account here (think about at least) is what happens if the slurm job gets scheduled/transitions to running while we are away. are we going to miss the resources running message? (i can't remember)

Reply:
We get Slurm states via polling the job status, not from state in the DispatcherRM.     So we can't miss RUNNING, unless it has already FAILED/COMPLETED by the time we re-attach, and in that case we don't care.

In any case send start_time never gets set for Slurm jobs so it isn't going to help us :-)